ceph_deploy/osd.py

```diff
@@ -706,46 +767,81 @@ def make_disk(parser):
     """
     Manage disks on a remote host.
     """
-    parser.add_argument(
-        'subcommand',
-        metavar='SUBCOMMAND',
-        choices=[
-            'list',
-            'prepare',
-            'activate',
-            'zap',
-            ],
-        help='list, prepare, activate, zap',
+    disk_parser = parser.add_subparsers(dest='subcommand')
+
+    disk_zap = disk_parser.add_parser(
+        'zap',
+        help='destroy existing partition table and content for DISK',
+        )
+    disk_zap.add_argument(
+        'disk',
+        nargs='+',
+        metavar='HOST:DISK',
+        type=colon_separated,
+        help='host and disk'
+        )
+
+    disk_list = disk_parser.add_parser(
+        'list',
+        help='List disk info from remote host(s)'
         )
-    parser.add_argument(
+    disk_list.add_argument(
         'disk',
         nargs='+',
         metavar='HOST:DISK',
         type=colon_separated,
-        help='host and disk (or path)',
+        help='remote host to list OSDs from'
         )
-    parser.add_argument(
+
+    disk_prepare = disk_parser.add_parser(
+        'prepare',
+        help='Prepare a disk for use as Ceph OSD by formatting/partitioning disk'
+        )
+    disk_prepare.add_argument(
         '--zap-disk',
         action='store_true', default=None,
         help='destroy existing partition table and content for DISK',
         )
-    parser.add_argument(
+    disk_prepare.add_argument(
         '--fs-type',
         metavar='FS_TYPE',
+        choices=['xfs',
+                 'ext4',
+                 'btrfs'
+                 ],
         default='xfs',
-        help='filesystem to use to format DISK (xfs, btrfs or ext4)'
+        help='filesystem to use to format DISK',
```

Comment:
I'm happy to put it back.  It's probably more useful that way.

I removed it because I added the `choices` field to enforce only the filesystems we allow. If you provide a bad one now, it will show:

```
$ ceph-deploy disk prepare --fs-type zfs
usage: ceph-deploy disk prepare [-h] [--zap-disk] [--fs-type FS_TYPE]
                                [--dmcrypt] [--dmcrypt-key-dir KEYDIR]
                                HOST:DISK [HOST:DISK ...]
ceph-deploy disk prepare: error: argument --fs-type: invalid choice: 'zfs' (choose from 'xfs', 'ext4', 'btrfs')
```

Previously the argument parser wouldn't kick this out, we would raise our own exception, which isn't necessary (we can fail sooner via argparse).

Howevever, you would still have to run the command with a bad fstype to see the list.  So I'll put it back.  And now that I've looked closer, I will remove the following redundant code:

```
if fs_type not in ('btrfs', 'ext4', 'xfs'):
            raise argparse.ArgumentTypeError...
```